packages/feedback-react/src/useFeedback.ts

```diff
@@ -0,0 +1,58 @@
+import { useState, useRef, useCallback, useEffect } from "react";
+import { FeedbackType, FeedbackOption } from "./types";
+
+export const useFeedback = (onSubmit: (f: FeedbackType) => void) => {
+    const [value, setValue] = useState<FeedbackOption>();
+    const [message, setMessage] = useState<string>();
+    const [submitted, setSubmitted] = useState(false);
+
+    const feedbackRef = useRef({
```

Comment:
Hvorfor brukes useRef her? Hele objektet ser egentlig unødvendig ut.

Reply:
Jeg så kommentarene i fila useFollowup nå, om autosubmit ved unmount.
Jeg forstår fortsatt ikke at det skal være nødvendig å legge verdiene i et objekt på en ref. Verdiene er vel tilgjengelige like lenge som refen er tilgjengelig(!?), så hvilken forskjell gjør refen?

Reply:
Det skulle man tro! Men vi måtte legge det til på dagens løsning for at det ikke skulle bli krøll. Tror det har med rekkefølgen ting oppdateres i å gjøre, men det ble i hvert fall ikke med riktige verdier uten å mellomlagre dem. Kan sikkert se litt nærmere på det nå som jeg sitter med komponenten uansett 😊

Reply:
Det manglet ikke noen deps i arrayet til effekten?
Veldig interessant om vi kan komme til bunns i dette :smiley:

Reply:
Om jeg ikke husker feil så var dette i forbindelse med å få sendt eventet når man unmounter komponenten. States klarte ikke å kjøre oppdateringen tidlig nok til at eventet faktisk fikk propagert ut av komponten, mens med en ref som er oppdatert på øyeblikket lot det seg gjøre.

Vi prøvde ganske mange forskjellige varianter uten å klare å få det til å fungere :/

Reply:
Hvilken state er det vi trenger at skal oppdatere seg? Verdiene på objektet på refen er jo basert på states (pluss et parameter). Hvis disse ikke er oppdatert, så er ikke refen heller oppdatert :thinking: